lib/puppet/application/lookup.rb

```diff
@@ -57,6 +57,102 @@ class Puppet::Application::Lookup < Puppet::Application
     end
   end
 
+  def help
+    <<-'HELP'
+
+puppet-lookup(8) -- Data in modules lookup function
+========
+
+SYNOPSIS
+--------
+The lookup command is used for debugging and testing a given data
+configuration. For a given data key, lookup will return which value
+will be produced with an optional explanation of how the system arrived
+at that value.
+
+USAGE
+-----
+puppet lookup [--help] [--type <TYPESTRING>] [--merge unique|hash|deep]
+  [--knock_out_prefix <PREFIX-STRING>] [--sort_merged_arrays]
+  [--unpack_arrays <STRING-VALUE>] [--merge_hash_arrays] [--explain]
+  [--default <VALUE>] [--node <NODE-NAME>] [--facts <FILE>] <keys>
+
+DESCRIPTION
+-----------
+The lookup command is a CLI interface for the puppet lookup function.
+When given one or more keys, the lookup command will return the first
+value found. The only component that is required in addition to the
+keys is a node (passed in via --node) which specifies which scope to
+perform the lookup in. It will exit with 0 if a value was returned
+and 1 otherwise.
+
+The other options are as passed into the lookup function, and the effect
+they have on the lookup is described in more detail in the header
+for the lookup function:
+
+https://github.com/puppetlabs/puppet/blob/master/lib/puppet/functions/lookup.rb
```

Comment:
This is problematic as a link since it refers to the latest version (i.e, head on master). Maybe it is best to point to the documentation site using a google link (like we do for deprecations). Text should probably say something like "look at the documentation for the loopup() function for the version of puppet you are using" - with a link to the documentation page (via an indirect google link so we can update it). Also log a ticket to review that link once we are close to release to ensure that it works). (The other alternative is to include the documentation here (copy/paste/edit) from the lookup function, but that means we have to remember to make updates in two places.)